fmlcore/src/main/java/net/minecraftforge/fml/ModWorkManager.java

```diff
@@ -23,8 +23,9 @@ public interface DrivenExecutor extends Executor {
 
         default void drive(Runnable ticker) {
             if (!selfDriven()) {
-                while (driveOne()) {
-                    ticker.run();
+                ticker.run();
+                while (true) {
+                    if (!driveOne()) break;
```

Comment:
Out of curiosity, what's the reason for this logic change (no longer running the ticker once for every completed task)? If the queue is full it seems it could go a while without being run.

Reply:
It's called way more often than there are tasks to process. Throttling to only when there's an active task is one of the key causes of no movement.